### ISSUE TITLE
HTML: TOC: Smooth scrolling

### ISSUE THREAD

**Issue by tim-gromeyer** [doxygen/doxygen#9855]
**Describe the error**.
When clicking on a section/link in the table of contents, it does not scroll smoothly. When clicking in the navigation bar, it scrolls smoothly.

**Expected behavior**
Smooth scrolling as when clicking a section in the navigation bar.

**To Reproduce**
[reproduce.zip](https://github.com/doxygen/doxygen/files/10726034/reproduce.zip)

**Version**
Doxygen version: `1.9.6 (4586b5cfaa3d46d51f6a51882951d15644c49edf)` (from release)
System:
```
Distributor ID:	Linuxmint
Description:	Linux Mint 21.1
Release:	21.1
Codename:	vera
```

**Comment by albert-github** [comment]
When in the clicking in the navigation tree:
![image](https://user-images.githubusercontent.com/5223533/218707972-d6c22a01-0fc3-4524-9bd1-2bbb2f99f364.png)
on "Another section" it "smoothly" scrolls to the section.
When clicking in the "Table of Contents":
![image](https://user-images.githubusercontent.com/5223533/218708193-bebe6695-cdfe-483c-b05d-354179c95918.png)
on "Another section" it jumps to the section. The later also occurs when clicking on "Section 2" in the running text:
![image](https://user-images.githubusercontent.com/5223533/218708438-1119e93f-df51-492e-8465-e7d42631765e.png)

This is not an easy thing to solve (as already remarked in https://github.com/jothepro/doxygen-awesome-css/issues/107) and the question is which parts should be affected:
- just the "Table of contents"
- all hyperlinks
- ...

@doxygen any idea how / where to accomplish this?

**Comment by tim-gromeyer** [comment]
> ... and the question is which parts should be affected:
> 
> * just the "Table of contents"
> * all hyperlinks
> * ...

I think it would be good to rework the whole section linking. It is complicated to link to a section because it uses the label `autotoc_md` + the section number. So if you add another section before the section you link to, you have to update all the links because the section you want to link to now has a different number.

In summary, I would say that reworking the table of contents alone is not enough.

Ideas other than a reworking are welcome.

**Comment by albert-github** [comment]
The `autotoc_md` are not for general use, just for internal use. When one wants to use a label one either has to use commands like `\section` and `\anchor` or in the markdown case use the explicit label as described in https://www.doxygen.nl/manual/markdown.html#md_header_id

So I don't think reworking labels is necessary.

**Comment by doxygen** [comment]
Please verify if the referenced commit fixes the problem (implemented it for the TOC links as these are also in the navigation panel).

**Comment by tim-gromeyer** [comment]
It fixed it, thank you!

**Comment by albert-github** [comment]
Solution has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
